pennylane/circuit_graph.py

```diff
@@ -522,12 +522,16 @@ def greedy_layers(self, wire_order=None, show_all_wires=False):
                     observables[wire] = [None]
         else:
 
-            # There are multiple measurements
-            for wire in sorted(self._grid):
-                mp_map = dict(zip(self.observables, range(self.max_simultaneous_measurements)))
+            # There are wire(s) with multiple measurements.
+            # We are creating a separate "visual block" at the end of the
+            # circuit for each observable and mapping observables with block
+            # indices.
+            num_observables = len(self.observables)
```

Comment:
This is problematic when we introduce `qml.sample()`: 

<img width="1044" alt="image" src="https://user-images.githubusercontent.com/21964864/135333871-5975061a-f507-4fc6-97f8-9385e133283e.png">

I think the `qml.sample()` calls don't register as observables so they aren't mapped to an index.

Reply:
Seems to be working now !